src/include/simeng/branchpredictors/GenericPredictor.hh

```diff
@@ -71,6 +84,13 @@ class GenericPredictor : public BranchPredictor {
 
   /** The size of the RAS. */
   uint16_t rasSize_;
+
+  // This variable is used only in debug mode -- therefore hide behind ifdef
```

Comment:
Not having the ifdef wouldn't cause an error though? I'm not even sure it would cause a warning given its a class member variable

Reply:
It does cause warnings with Clang which are now errors due to Werror - unused variable when in release mode. Solutions such as marking as [[maybe_unused]] cause warnings which are now errors in GCC as in debug mode it is used. This seems to me like GCC shouldn't actually be complaining about that, but it does, so we are kind of stuck. This seemed like the most sensible solution. Ideally the comment would say all of this to prevent this confusion. Any suggestions on saying all of that concisely

Reply:
Potentially the updated comment is confusing as well. I wasn't specific enough about the wording, my bad. Should be saying "hide behind the ifdef to prevent errors [because above]"

Reply:
I agree that comment could do more.  I've updated to explain the context for this solution

Reply:
I see, thats fair enough. I'd agree then that the comment above the `ifdef` needs to explain the above

Reply:
Comment updated

Reply:
Moved to BranchPredictor.hh. I'm happy with the updated comment but leaving for @FinnWilkinson to resolve